cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -538,28 +574,44 @@ protected ContextDefinitions readContextDefinitions(String path) throws Exceptio
      *         between libraries (e.g. LibraryName.ExpressionName).
      */
-    protected Tuple2<Object, Map<String, Object>> evaluate(Tuple2<Object, List<Row>> rowsByContext,
-            String contextName, CqlEvaluator evaluator, CqlEvaluationRequests requests, SparkOutputColumnEncoder columnEncoder, LongAccumulator perContextAccum) {
+    protected Tuple2<Object, Map<String, Object>> evaluate(Tuple2<Object,
+                                                           List<Row>> rowsByContext,
+                                                           String contextName,
+                                                           CqlEvaluator evaluator,
+                                                           CqlEvaluationRequests requests,
+                                                           SparkOutputColumnEncoder columnEncoder,
+                                                           LongAccumulator perContextAccum,
+                                                           CollectionAccumulator<EvaluationError> errorAccum) {
         perContextAccum.add(1);
         
         List<CqlEvaluationRequest> requestsForContext = requests.getEvaluationsForContext(contextName);
         
         Map<String, Object> expressionResults = new HashMap<>();
         for (CqlEvaluationRequest request : requestsForContext) {
-            try {
-                CqlEvaluationResult result = evaluator.evaluate(request, args.debug ? CqlDebug.DEBUG : CqlDebug.NONE);
-                for (Map.Entry<String, Object> entry : result.getExpressionResults().entrySet()) {
-                    String outputColumnKey = columnEncoder.getColumnName(request, entry.getKey());
-                    expressionResults.put(outputColumnKey, typeConverter.toSparkType(entry.getValue()));
+            for (CqlExpressionConfiguration expression : request.getExpressions()) {
+                CqlEvaluationRequest singleRequest = new CqlEvaluationRequest(request);
+                singleRequest.setExpressions(Collections.singleton(expression));
+                try {
+                    CqlEvaluationResult result = evaluator.evaluate(singleRequest, args.debug ? CqlDebug.DEBUG : CqlDebug.NONE);
+                    for (Map.Entry<String, Object> entry : result.getExpressionResults().entrySet()) {
+                        String outputColumnKey = columnEncoder.getColumnName(request, entry.getKey());
+                        expressionResults.put(outputColumnKey, typeConverter.toSparkType(entry.getValue()));
+                    }
+                } catch (Throwable th) {
+                    if (errorAccum != null) {
+                        Object contextId = rowsByContext._1();
+                        errorAccum.add(new EvaluationError(contextName, contextId, singleRequest.getExpressionNames().iterator().next(), th.getMessage()));
+                    }
+                    else {
+                        throw new RuntimeException("CQL evaluation failed: ", th);
```

Comment:
The new exception is not as verbose as the original.
Any particular reason to simplify the error message?

Reply:
We don't want context keys making their way to LogDNA.  I suppose I could still report the context name and the request so there is some additional info.